cli/spec.py

```diff
@@ -16,30 +20,187 @@
 TRIM_PARAMS = ['default', 'required']
 
 
-def cfg_file_to_dict(file_path):
+class ValueArgument(object):
+    """
+    Default argument type for InfraRed Spec
+    Resolves "None" Types with priority defaults.
+    """
+    settings_dir = None
+    subcommand = None
+
+    def __init__(self, value=None, required=None):
+        self.value = value
+        self.arg_name = None
+        self.required = None
+
+    @classmethod
+    def get_app_attr(cls, attribute):
```

Comment:
I'm not sure how to do that.
This method is setting and getting class attributes, but it's called from both classmethods and instance methods, so the only way I know to do that is with a classmethod